### PR TITLE
Yaml: Removed boolean flags

### PR DESCRIPTION
**Pull request by chr-hertel**
Remove deprecated boolean flags and use const instead.

### CODE REVIEW

File: components/yaml.rst [hunk 1]
```diff
@@ -224,20 +224,20 @@ Instead of encoding as ``null`` you can choose to throw an exception if an inval
...
 
     // throw an exception if a resource or object is encountered
-    Yaml::dump($data, 2, 4, true);
+    Yaml::dump($data, 2, 4, Yaml::DUMP_EXCEPTION_ON_INVALID_TYPE);
 
     // throw an exception if an encoded object is found in the YAML string
-    Yaml::parse($yaml, true);
+    Yaml::parse($yaml, Yaml::PARSE_EXCEPTION_ON_INVALID_TYPE);
 
 However, you can activate object support using the next argument::
```
Comment (wouterj) [inline components/yaml.rst]: this is no longer true, it's not the next statement. Can you please rewrite this sentence?

status: needs work

**Comment by wouterj** [comment]
Thanks for catching this and fixing it! I've left one minor comment. If you can fix it, it would be great. Otherwise, we'll do it while merging.

Status: needs work

**Comment by xabbuh** [comment]
For the records: This will fix parts of #6226.

**Comment by chr-hertel** [comment]
@wouterj sorry, missed that
@xabbuh i'll try to have a look at #6226 the next days

**Comment by wouterj** [comment]
> @wouterj sorry, missed that

No worries, you're doing a perfect job! :)

File: components/yaml.rst [hunk 2]
```diff
@@ -214,6 +228,42 @@ changed using the third argument as follows::
             foo: bar
             bar: baz
 
+Mulit-line strings as scalar blocks
```
Comment (javiereguiluz) [inline components/yaml.rst]: `Mulit-line` -> `Multi-line`

File: components/yaml.rst [hunk 3]
```diff
@@ -214,6 +228,42 @@ changed using the third argument as follows::
...
+...................................
+
+To dump multi-line strings including ``\n`` character to a scalar block
+in YAML, just use the ``DUMP_MULTI_LINE_LITERAL_BLOCK`` flag::
```
Comment (javiereguiluz) [inline components/yaml.rst]: `just use` -> `use`

File: components/yaml.rst [hunk 4]
```diff
@@ -214,6 +228,42 @@ changed using the third argument as follows::
...
+
+    content: |
+        Lorem ipsum dolor sit amet
+        consectetur adipisicing
+        elit sed do eiusmod
+
+Dump Objects as YAML Maps
+.........................
+
+You can easily dump PHP Objects as YAML map by passing ``DUMP_OBJECT_AS_MAP`` flag.
```
Comment (javiereguiluz) [inline components/yaml.rst]: This line needs to end with `::` to activate the highlight of the following PHP code block.

**Comment by javiereguiluz** [comment]
@chr-hertel I left some minor comments but your pull request is great! Thank you.

👍 

Status: reviewed

**Comment by chr-hertel** [comment]
thanks @javiereguiluz, i fixed that.

**Comment by xabbuh** [comment]
@chr-hertel Sorry the late feedback: It looks like we covered parts of this with #6622. Can you rebase here and resolve the conflicts?

Status: Needs work

**Comment by HeahDude** [comment]
Hello @chr-hertel do you still have time to work on this or do you want us to finish it? Thanks!

**Comment by slde-thespirit** [comment]
i guess this one is not longer needed.

**Pull request closed without merging**